filter.php

```diff
@@ -244,7 +258,7 @@ private function apply_loadonvisible(array $match, stored_file $file, $originals
         if (!$imageinfo || !isset($imageinfo->width)) {
             return ($img);
         }
-        $width = $imageinfo->width;
+        $width = $imageinfo->width > $maxwidth ? $maxwidth : $imageinfo->width;
         $height = $imageinfo->height;
```

Comment:
If we go over the maximum width then we should adjust the height in ratio to the adjusted width.

Reply:
Why calculating max width in this function when you do not use it?

But yes, the height should be also adjusted.

Reply:
It will take me a while to familiarise myself with the code before I can answer that - I wrote it about 5 years ago!

Reply:
OK - I've had a look at the code base and this is what I've found:

apply_loadonvisible only replaces the "<img" portion which should leave any width / height attributes intact.

The maxwidth variable as you stated is not used - it should actually be removed. That's the fix we should go with.

The maxwidth is actually handled in img_add_width_height so we don't need to worry about it in apply_loadonvisible.